regression_and_census.R

```diff
@@ -9,9 +9,169 @@
+
+
+# Clean and transform data for California ---------------------------------
+
+#merging the two datasets
+pol_covid_census_2020 <- CA_PM25_wk_PM %>%
+  left_join(census_counties_2020, by=c("county"="BASENAME")) %>%
+  # creating pop density variable from census data
+  mutate(pop_density_new=POP100 /AREALAND)
+
+
+# explanatory and response variables
+explanatory_var <- "PM25_wk_avg"
+# This line checks the column names in CA_PM25_wk_PM to identify which columns
+# are response variables: those with 'lag_wkg' in the colum name)
+response_vars <- grep("lag_wkavg", names(CA_PM25_wk_PM), value=TRUE)
+covariates <- c("month", "pop_density_new", "P0020002", "P0020003", "P0020005",
+                "P0020006", "P0020007", "P0020008", "P0020009", "P0020010")
+
+
+# Create datasets to fit models to ----------------------------------------
+
+#UPDATE - data with cutoff at 30 PM
+PM_CA_all <- pol_covid_census_2020 %>% 
+  # Month should be a categorical variable, hence changing its type numeric --> factor.
+  mutate(month=factor(month)) %>%
+  filter(., complete.cases(.)) %>%
+  slice(-unique(which(select(., c(explanatory_var, response_vars, covariates)) < 0, arr.ind=TRUE)[, 1]))
+PM_CA_all[, response_vars] %<>% sapply(., as.integer)
+cols_to_standardize <- c("pop_density_new", grep("P00", covariates, value=TRUE))
+multipliers <- select(PM_CA_all, cols_to_standardize) %>% apply(2, max)
+PM_CA_all[, cols_to_standardize] %<>%
+  sapply(., function (x) x/max(x))
+
+PM_CA_below30 <- filter(PM_CA_all, PM25_wk_avg <= 30)
+PM_CA_above30 <- filter(PM_CA_all, PM25_wk_avg > 30)
+
+
+plot(PM_CA_below30$PM25_wk_avg, PM_CA_below30$case3lag_wkavg)
+
+
+# Run models --------------------------------------------------------------
+
+# Structure of this section
+# 1. `run_model` function to run a model and return its results in a user-friendly object.
+# 2. `get_model_results` function to call the `run_model` function on a specific model 
+#     configuration, e.g lm, PM < 30, with/without covariates.
+# 3. A list defining the model configurations.
+# 4. Run models and create output tables
+
+
+# 1. Function to run a model and return its results in a user-friendly object.
+
+run_model <- function (model_func_str, model_name, dataset, ycol, xcol, other_args=NULL) {
+  # Creates the call to the model fitting function, fit the model to data, and 
+  # return the results in a user-friendly table.
+  formula_str <- paste(ycol, "~", paste(xcol, collapse="+"))
+  formula_obj <- as.formula(formula_str)
+  model_func <- eval(parse(text=model_func_str))
+  if (model_func_str=="geeglm") {
+    model <- model_func(formula=formula_obj, data=dataset, na.action=na.omit, id=county, family=gaussian, corstr="ar1")
+  } else {
+    model <- model_func(formula=formula_obj, data=dataset, na.action=na.omit)
+  }
+  results_df <- broom::tidy(model, conf.int=TRUE)[2, c("estimate", "conf.low", "conf.high", "p.value")]
```

Comment:
The `tidy` function from the `broom` package is a useful function for turning the model output into a user-friendly data frame.